parsec/backend/client_context.py

```diff
@@ -27,13 +27,21 @@
 
 
 class BaseClientContext:
-    __slots__ = ("conn_id", "api_version")
+    __slots__ = ("conn_id", "api_version", "_cancel_scope")
     TYPE: ClientType
     logger: BoundLogger
 
     def __init__(self, api_version: ApiVersion):
         self.api_version = api_version
         self.conn_id = uuid4().hex
+        self._cancel_scope: Optional[trio.CancelScope] = None
+
+    def set_scope(self, client_scope: trio.CancelScope) -> None:
```

Comment:
remove this method and instead directly use `self.connection_cancel_scope: Optional[trio.CancelScope]`

providing a setter is often a good idea for encapsulation, but here we already directly modify client_ctx attribute:
https://github.com/Scille/parsec-cloud/blob/9f42cfb69edfe34b92c8cc2b9f80707ac07e5571/parsec/backend/asgi/ws.py#L69

So it would be strange to have two way of modifying the attribute, and we don't really need encapsulation here given how simple this object is (and we are not in Java so there is no jail time for doing this anyway :trollface: )